File: user-service/src/main/java/com/example/user/services/UserService.java
package com.example.user.services;

import com.example.user.client.TaskServiceClient;
import com.example.user.client.entities.Task;
import com.example.user.model.User;
import com.example.user.model.UserTask;
import com.example.user.repositories.UserRepository;
import com.example.user.repositories.entities.UserEntity;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.List;

@Singleton
public class UserService {
    private final UserRepository userRepository;
    private final TaskServiceClient taskServiceClient;

    @Inject
    public UserService(UserRepository userRepository, TaskServiceClient taskServiceClient) {
        this.userRepository = userRepository;
        this.taskServiceClient = taskServiceClient;
    }

    public List<User> getAll() {
        return userRepository.findAll().stream()
                .map(e -> new User(e.getId(), e.getName(), e.getTaskIds()))
                .toList();
    }

    public User save(User user) {
        UserEntity savedUserEntity = new UserEntity(null, user.name(), user.taskIds());
        savedUserEntity = userRepository.save(savedUserEntity);

        return new User(
                savedUserEntity.getId(),
                savedUserEntity.getName(),
                savedUserEntity.getTaskIds()
        );
    }

    public void updateTaskList(String userId, Integer taskId) {
        UserEntity userEntity = userRepository.findById(userId).orElseThrow();
        userEntity.getTaskIds().add(taskId.longValue());
        userRepository.update(userEntity);
    }

    public List<UserTask> getAllUsersTasks() {
        List<User> users = getAll();
        List<Task> tasks = taskServiceClient.getAll();

        return users.stream()
                .map(user -> new UserTask(
                                user.id(),
                                user.name(),
                                tasks.stream()
                                        .filter(task -> user.taskIds().contains(task.id()))
                                        .map(Task::description)
                                        .toList()
                        )
                )
                .toList();
    }
}

File: user-service/app/services/user_service.py
from ..repository.user_repository import UserRepository
from ..clients.task_service_client import task_service_client
from ..dto.user_task import UserTask

class UserService:
    def __init__(self):
        self.user_repository = UserRepository()
        self.tasks_service_client = task_service_client
    
    def get_all_users_tasks(self):
        users = self.user_repository.get_all()
        tasks = self.tasks_service_client.get_all()
        
        task_map = {task['id']: task['description'] for task in tasks}

        return [
            UserTask(
                id=user['id'],
                name=user['name'],
                task_descriptions=[
                    task_map[task_id]
                    for task_id in user.get('task_ids', [])
                    if task_id in task_map
        ],
            )
            for user in users
        ]        

user_service = UserService()